docs(testing): add componentOnReady guidance and a Vue testing page (#4611)
ionic-team/ionic-docs: docs/angular/testing.md

});
```

> ### Import `componentOnReady` for standalone projects

Reviewer: I recommend shortening the title to match the title standards.

```suggestion
### Waiting for Components
```

> ### Import `componentOnReady` for standalone projects

Reviewer: Move this to the end of `### Pages and Components`, under `## Unit Testing`, because that's where component DOM testing lives and this follows on from that section's closing paragraph about `fixture.nativeElement.querySelector`. Keep it at h3 so it still shows up in the page's Contents sidebar.

> When testing Ionic components, use the exported `componentOnReady` helper from `@ionic/core` instead of calling `el.componentOnReady()` directly. The helper works with both lazy-loaded and custom-element builds, making it more likely the component has finished rendering before making assertions against its rendered DOM or running accessibility tests. See an example here:

Reviewer: "More likely" hides a real difference: the lazy-loaded path awaits an actual readiness promise while the custom-element path is just one animation frame, so let's state both paths and the reason the direct call fails.

```suggestion
When testing Ionic components, use the `componentOnReady` helper exported from `@ionic/core` rather than calling `el.componentOnReady()` directly. That method only exists on lazy-loaded elements. Calling it directly throws on custom-element builds, which is what standalone projects use. The helper handles both. It awaits the element's own `componentOnReady()` promise when that exists. Otherwise it waits one animation frame, giving the component's inner contents a chance to render. Wait for the callback before asserting against the rendered DOM or running accessibility tests.
```

```tsx
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { axe, toHaveNoViolations } from 'jasmine-axe';
import { componentOnReady } from '@ionic/core';

import { Component } from './component';

describe('Component', () => {
  let component: Component;
  let fixture: ComponentFixture<Component>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [Component],
    }).compileComponents();

    fixture = TestBed.createComponent(Component);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should pass accessibility test', async () => {
    const el = fixture.nativeElement.querySelector('ion-button');

    await new Promise<void>((resolve) => {
      componentOnReady(el, () => resolve());
    });

    jasmine.addMatchers(toHaveNoViolations);

    const a11y = await axe(fixture.nativeElement);

    expect(a11y).toHaveNoViolations();
  });
});

> ```

Reviewer: Trimmed to just the readiness wait so the three lines that matter are visible; jasmine-axe and the unused component variable are a separate concern, and HomePage reads better than Component next to Angular's own decorator.

```suggestion
```tsx
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { componentOnReady } from '@ionic/core';

import { HomePage } from './home.page';

describe('HomePage', () => {
  let fixture: ComponentFixture<HomePage>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [HomePage],
    }).compileComponents();

    fixture = TestBed.createComponent(HomePage);
    fixture.detectChanges();
  });

  it('renders the submit button', async () => {
    const button = fixture.nativeElement.querySelector('ion-button');

    await new Promise<void>((resolve) => componentOnReady(button, () => resolve()));

    expect(button.textContent).toContain('Submit');
  });
});
```
```
